network/src/p2p.rs

```diff
@@ -296,3 +317,43 @@ impl ReliableNetwork<WorkerMessage> for P2pNetwork {
         self.send(peer, f).await
     }
 }
+
+#[async_trait]
+impl UnreliableNetwork<WorkerBatchRequest> for P2pNetwork {
+    type Response = WorkerBatchResponse;
+    async fn unreliable_send(
+        &mut self,
+        peer: NetworkPublicKey,
+        message: &WorkerBatchRequest,
+    ) -> JoinHandle<anyhow::Result<anemo::Response<WorkerBatchResponse>>> {
+        let message = message.to_owned();
+        let f = move |peer| async move {
+            WorkerToWorkerClient::new(peer)
+                .request_batches(message)
+                .await
+        };
+        self.unreliable_send(peer, f).await
+    }
+}
+
+#[async_trait]
+impl ReliableNetwork<WorkerBatchRequest> for P2pNetwork {
+    type Response = WorkerBatchResponse;
+    async fn send(
+        &mut self,
+        peer: NetworkPublicKey,
+        message: &WorkerBatchRequest,
+    ) -> CancelOnDropHandler<anyhow::Result<anemo::Response<WorkerBatchResponse>>> {
+        let message = message.to_owned();
+        let f = move |peer| {
+            let message = message.clone();
+            async move {
+                WorkerToWorkerClient::new(peer)
+                    .request_batches(message)
```

Comment:
I'm a bit uncomfortable with `request_batches` being the method that sends a generic network message between workers. Could we do a bit of renaming?